2025-05-17 17:13:17,886 INFO    Thread-3 (_run_job):778004 [wandb_init.py:setup_run_log_directory():662] Logging user logs to /speech/utkarsh/da6401_assignment3/wandb/run-20250517_171317-3vgfukgh/logs/debug.log
2025-05-17 17:13:17,886 INFO    Thread-3 (_run_job):778004 [wandb_init.py:setup_run_log_directory():663] Logging internal logs to /speech/utkarsh/da6401_assignment3/wandb/run-20250517_171317-3vgfukgh/logs/debug-internal.log
2025-05-17 17:13:17,886 INFO    Thread-3 (_run_job):778004 [wandb_init.py:init():781] calling init triggers
2025-05-17 17:13:17,886 INFO    Thread-3 (_run_job):778004 [wandb_init.py:init():786] wandb.init called with sweep_config: {'batch_size': 32, 'beam_size': None, 'cell_type': 'lstm', 'decoder_dropout': 0.3, 'embedding_size': 256, 'encoder_dropout': 0.4, 'hidden_size': 128, 'learning_rate': 0.001, 'num_decoder_layers': 3, 'num_encoder_layers': 1}
config: {'data_path': 'dakshina_dataset_v1.0/hi/lexicons/', 'embedding_size': 256, 'hidden_size': 128, 'num_encoder_layers': 1, 'num_decoder_layers': 3, 'encoder_dropout': 0.4, 'decoder_dropout': 0.3, 'cell_type': 'lstm', 'batch_size': 32, 'learning_rate': 0.001, 'n_epochs': 20, 'clip': 1.0, 'teacher_forcing_ratio': 0.5, 'patience': 5, 'min_delta': 0.0, 'beam_size': None, 'save_path': 'models/', 'model_name': 'best_model.pt', 'seed': 42, 'log_wandb': True, 'wandb_project': 'hindi-transliteration', 'wandb_name': 'sweep-3vgfukgh', '_wandb': {}}
2025-05-17 17:13:17,886 INFO    Thread-3 (_run_job):778004 [wandb_init.py:init():801] wandb.init() called while a run is active
2025-05-17 17:13:37,479 INFO    Thread-3 (_run_job):778004 [wandb_run.py:_finish():2189] finishing run da24s011-indian-institute-of-technology-madras/hindi-transliteration/3vgfukgh
2025-05-17 17:13:37,479 INFO    Thread-3 (_run_job):778004 [wandb_run.py:_atexit_cleanup():2419] got exitcode: 1
2025-05-17 17:13:37,479 INFO    Thread-3 (_run_job):778004 [wandb_run.py:_restore():2401] restore
2025-05-17 17:13:37,479 INFO    Thread-3 (_run_job):778004 [wandb_run.py:_restore():2407] restore done
2025-05-17 17:13:38,732 INFO    Thread-3 (_run_job):778004 [wandb_run.py:_footer_history_summary_info():4064] rendering history
2025-05-17 17:13:38,732 INFO    Thread-3 (_run_job):778004 [wandb_run.py:_footer_history_summary_info():4096] rendering summary
2025-05-17 17:13:38,732 INFO    Thread-3 (_run_job):778004 [wandb_run.py:_footer_sync_info():4025] logging synced files
2025-05-17 17:13:38,733 ERROR   MainThread:778004 [pyagent.py:_run_jobs_from_queue():234] [no run ID] Run 3vgfukgh errored:
Traceback (most recent call last):
  File "/speech/utkarsh/.local/lib/python3.10/site-packages/wandb/agents/pyagent.py", line 306, in _run_job
    self._function()
  File "/speech/utkarsh/da6401_assignment3/sweep.py", line 44, in sweep_train
    main(model_config)
  File "/speech/utkarsh/da6401_assignment3/main.py", line 80, in main
    history = train_model(
  File "/speech/utkarsh/da6401_assignment3/train.py", line 211, in train_model
    train_loss = train(model, train_iterator, optimizer, criterion, clip, device)
  File "/speech/utkarsh/da6401_assignment3/train.py", line 42, in train
    output = model(src, trg, teacher_forcing_ratio)
  File "/speech/utkarsh/.local/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1110, in _call_impl
    return forward_call(*input, **kwargs)
  File "/speech/utkarsh/da6401_assignment3/model.py", line 233, in forward
    output, hidden = self.decoder(input, hidden)
  File "/speech/utkarsh/.local/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1110, in _call_impl
    return forward_call(*input, **kwargs)
  File "/speech/utkarsh/da6401_assignment3/model.py", line 169, in forward
    output, hidden = self.rnn(embedded, hidden)
  File "/speech/utkarsh/.local/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1110, in _call_impl
    return forward_call(*input, **kwargs)
  File "/speech/utkarsh/.local/lib/python3.10/site-packages/torch/nn/modules/rnn.py", line 759, in forward
    self.check_forward_args(input, hx, batch_sizes)
  File "/speech/utkarsh/.local/lib/python3.10/site-packages/torch/nn/modules/rnn.py", line 685, in check_forward_args
    self.check_hidden_size(hidden[0], self.get_expected_hidden_size(input, batch_sizes),
  File "/speech/utkarsh/.local/lib/python3.10/site-packages/torch/nn/modules/rnn.py", line 226, in check_hidden_size
    raise RuntimeError(msg.format(expected_hidden_size, list(hx.size())))
RuntimeError: Expected hidden[0] size (3, 32, 128), got [1, 32, 128]

2025-05-17 17:13:39,733 INFO    MsgRouterThr:778004 [mailbox.py:close():129] [no run ID] Closing mailbox, abandoning 1 handles.
